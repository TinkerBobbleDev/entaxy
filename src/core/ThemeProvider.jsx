import React from 'react'
import PropTypes from 'prop-types'
import { createMuiTheme, MuiThemeProvider } from '@material-ui/core/styles'
import CssBaseline from '@material-ui/core/CssBaseline'
import blueGrey from '@material-ui/core/colors/blueGrey'
import '../styles/globals.css'

const theme = createMuiTheme({
  palette: {
    primary: { main: blueGrey[800] },
    secondary: { main: '#1B9CC4' },
    info: {
      text: '#00acc1',
      background: '#00acc11a'
    },
    danger: {
      text: '#721c24',
      background: '#f8d7da66'
    }

  },
  typography: {
    useNextVariants: true
  }
})

const ThemeProvider = ({ children }) => (
  <CssBaseline>
    <MuiThemeProvider theme={theme}>
      {children}
    </MuiThemeProvider>
  </CssBaseline>
)

ThemeProvider.propTypes = {
  children: PropTypes.node.isRequired
}

export default ThemeProvider
