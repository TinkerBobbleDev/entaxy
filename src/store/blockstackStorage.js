/* eslint-disable no-console */
import { getFile, putFile, isUserSignedIn } from 'blockstack'

const filename = 'entaxy.json'

export const loadState = () => {
  if (isUserSignedIn()) {
    return getFile(filename)
      .then(data => JSON.parse(data))
      .catch((error) => {
        console.log(`Error loading file ${filename}:`, error)
        return {}
      })
  }
  return Promise.resolve({})
}

export const saveState = (state) => {
  if (isUserSignedIn()) {
    return putFile(filename, JSON.stringify(state))
  }
  return undefined
}
