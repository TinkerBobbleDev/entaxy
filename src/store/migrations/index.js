import migration0 from './0'
import migration1 from './1'
import migration2 from './2'
import migration3 from './3'

export default {
  0: migration0,
  1: migration1,
  2: migration2,
  3: migration3
}
