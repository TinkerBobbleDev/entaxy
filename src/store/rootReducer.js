import { combineReducers } from 'redux'
import userReducer from './user/reducer'
import settingsReducer from './settings/reducer'
import transactionsReducer from './transactions/reducer'
import marketValuesReducer from './marketValues/reducer'

export default combineReducers({
  user: userReducer,
  settings: settingsReducer,
  transactions: transactionsReducer,
  marketValues: marketValuesReducer
})
