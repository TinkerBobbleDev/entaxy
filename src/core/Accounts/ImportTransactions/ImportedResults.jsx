/* eslint-disable react/no-multi-comp */
import React from 'react'
import PropTypes from 'prop-types'
import { makeStyles } from '@material-ui/core/styles'
import Button from '@material-ui/core/Button'
import Divider from '@material-ui/core/Divider'
import CheckCircleIcon from '@material-ui/icons/CheckCircle'
import ErrorIcon from '@material-ui/icons/Error'
import Tooltip from '@material-ui/core/Tooltip'
import green from '@material-ui/core/colors/green'
import red from '@material-ui/core/colors/red'
import { Column } from 'react-virtualized'
import ResultsToolbar from './ResultsToolbar'
import TransactionsTable from '../Transactions/TransactionsTable'

const useStyles = makeStyles(() => ({
  tableWrapper: {
    height: 'calc(100vh - 350px)',
    marginBottom: 100
  },
  formActions: {
    display: 'flex',
    'justify-content': 'flex-end',
    'padding-top': '10px'
  },
  iconCheck: {
    color: green[500],
    marginRight: '8px',
    'vertical-align': 'text-bottom',
    fontSize: 18
  },
  iconError: {
    color: red[800],
    marginRight: '8px',
    'vertical-align': 'text-bottom',
    fontSize: 18
  }
}))

const ImportedResults = (props) => {
  const classes = useStyles()
  const {
    account,
    transactions,
    errors,
    onSave,
    onBack
  } = props

  const filterByErrors = transaction => (
    Object.keys(transaction).includes('errors') && transaction.errors.length > 0
  )

  const toolbarProps = () => {
    const transactionsWithErrors = transactions.filter(filterByErrors)
    let subTitle = 'No errors'

    if (transactionsWithErrors.length > 0) {
      subTitle = (
        <div>
          <ErrorIcon className={classes.iconError} />
          {transactionsWithErrors.length}
          &nbsp;transactions have errors and will not be imported
        </div>
      )
    }

    return {
      title: `Found ${transactions.length} transactions`,
      subTitle
    }
  }

  const errorCellRenderer = (data) => {
    if (data.cellData.length === 0) {
      return <CheckCircleIcon className={classes.iconCheck} />
    }
    return (
      <Tooltip title={data.cellData}>
        <ErrorIcon className={classes.iconError} />
      </Tooltip>
    )
  }

  return (
    <div>
      <TransactionsTable
        className={classes.tableWrapper}
        account={account}
        transactions={transactions}
        hideChekboxes
        Toolbar={ResultsToolbar}
        toolbarProps={toolbarProps()}
      >
        <Column
          width={20}
          disableSort={true}
          dataKey="errors"
          cellRenderer={errorCellRenderer}
        />
      </TransactionsTable>
      <Divider />
      <div className={classes.formActions}>
        <Button
          onClick={onSave}
          color="secondary"
          disabled={errors.base.length > 0}
        >
          Save Transactions
        </Button>
        <Button onClick={onBack} color="secondary">Back</Button>
      </div>
    </div>
  )
}

ImportedResults.propTypes = {
  account: PropTypes.object.isRequired,
  transactions: PropTypes.array.isRequired,
  errors: PropTypes.object.isRequired,
  onSave: PropTypes.func.isRequired,
  onBack: PropTypes.func.isRequired
}

export default ImportedResults
