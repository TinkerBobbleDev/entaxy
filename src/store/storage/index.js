/* eslint-disable import/no-cycle */
import { snakeCase } from 'lodash'
import * as blockstackStorage from './blockstackStorage'
import * as localStorage from './localStorage'
import { showOverlay, hideOverlay } from '../settings/actions'
import { initialState as settingsInitialState } from '../settings/reducer'
import { initialState as accountsInitialState } from '../accounts/reducer'
import { initialState as transactionsInitialState } from '../transactions/reducer'
import { initialState as exchangeRatesInitialState } from '../exchangeRates/reducer'
import { initialState as budgetInitialState } from '../budget/reducer'

export const loadItem = (itemName, newState) => (dispatch, getState) => {
  let payload = newState
  if (itemName === 'settings') {
    const { settings } = getState()
    if (settings !== null && settings !== undefined) {
      payload = { ...newState, overlayMessage: settings.overlayMessage }
    }
  }
  dispatch({
    type: `LOAD_${snakeCase(itemName).toUpperCase()}`,
    payload
  })
}

export const loadAllItems = (newState) => (dispatch) => {
  [
    'settings',
    'accounts',
    'transactions',
    'exchangeRates',
    'budget'
  ].map((itemName) => dispatch(loadItem(itemName, newState[itemName] || {})))
}


export const loadState = () => async (dispatch, getState) => {
  if (getState().user === undefined) return {}
  let newState
  switch (getState().user.isAuthenticatedWith) {
    case 'guest':
      dispatch(showOverlay('Loading data from localStorage...'))
      newState = await localStorage.loadState()
      break
    case 'blockstack':
      dispatch(showOverlay('Loading data from Blockstack...'))
      newState = await blockstackStorage.loadState()
      break
    // no default
  }
  if (newState === undefined) newState = { settings: settingsInitialState }
  dispatch(loadAllItems({
    ...newState,
    settings: {
      ...newState.settings,
      snackbarMessage: null
    }
  }))
  return dispatch(hideOverlay())
}

export const resetState = () => (dispatch, getState) => {
  const { settings } = getState()

  // Preserve the existing currency and locale
  dispatch(loadAllItems({
    settings: {
      ...settingsInitialState,
      currency: settings.currency,
      locale: settings.locale
    },
    accounts: accountsInitialState,
    transactions: transactionsInitialState,
    exchangeRates: exchangeRatesInitialState,
    budget: budgetInitialState
  }))
}

export const saveState = () => (_, getState) => {
  let { user } = getState()
  if (user === undefined) user = {}
  switch (user.isAuthenticatedWith) {
    case 'guest':
      return localStorage.saveState(getState())
    case 'blockstack':
      return blockstackStorage.saveState(getState())
    default:
      return undefined
  }
}
