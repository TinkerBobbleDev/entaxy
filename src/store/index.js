import { createStore, applyMiddleware } from 'redux'
import { composeWithDevTools } from 'redux-devtools-extension'
import thunkMiddleware from 'redux-thunk'
import rootReducer from './rootReducer'

const middlewares = [thunkMiddleware]

if (process.env.NODE_ENV === 'development') {
  const { logger } = require('redux-logger')
  middlewares.push(logger)
}

const store = createStore(
  rootReducer,
  composeWithDevTools(applyMiddleware(...middlewares))
)

if (module.hot) {
  module.hot.accept(rootReducer, () => {
    store.replaceReducer(rootReducer)
  })
}

export default store
