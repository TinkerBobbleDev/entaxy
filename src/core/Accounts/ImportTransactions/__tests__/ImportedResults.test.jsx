import React from 'react'
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux'
import store from '../../../../store'
import ThemeProvider from '../../../ThemeProvider'
import ImportedResults from '../ImportedResults'


describe('ImportedResults', () => {
  const mockOnSave = jest.fn()
  const mockOnBack = jest.fn()
  const account = {
    id: 1,
    name: 'Checking',
    institution: 'TD',
    currency: 'CAD'
  }
  const transactions = [{
    id: 1,
    accountId: 1,
    amount: 10,
    createAt: Date.now()
  }, {
    id: 2,
    accountId: 1,
    amount: -5,
    createAt: Date.now() + 10
  }]

  beforeEach(() => {
    jest.clearAllMocks()
  })

  it('matches snapshot without errors', () => {
    const component = renderer.create((
      <Provider store={store}>
        <ThemeProvider>
          <ImportedResults
            account={account}
            transactions={transactions}
            errors={{ base: [], transactions: [] }}
            onSave={mockOnSave}
            onBack={mockOnBack}
          />
        </ThemeProvider>
      </Provider>
    ))
    expect(component.toJSON()).toMatchSnapshot()
  })

  it('matches snapshot with errors', () => {
    transactions.push({
      id: 3,
      accountId: 1,
      amount: 11,
      createAt: Date.now() + 11,
      errors: ['some error']
    })
    const component = renderer.create((
      <Provider store={store}>
        <ThemeProvider>
          <ImportedResults
            classes={{ }}
            account={account}
            transactions={transactions}
            errors={{ base: [], transactions: [] }}
            onSave={mockOnSave}
            onBack={mockOnBack}
          />
        </ThemeProvider>
      </Provider>
    ))
    expect(component.toJSON()).toMatchSnapshot()
  })
})
