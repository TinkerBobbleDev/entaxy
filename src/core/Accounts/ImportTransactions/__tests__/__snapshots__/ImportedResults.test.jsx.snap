// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImportedResults matches snapshot with errors 1`] = `
<div>
  <div
    className="TransactionsTableComponent-tableWrapper-5 makeStyles-tableWrapper-1"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular TableToolbar-root-14 MuiToolbar-gutters"
    >
      <div
        className="TableToolbar-title-18"
      >
        <div>
          <h6
            className="MuiTypography-root MuiTypography-h6"
          >
            Found 3 transactions
          </h6>
          <span
            className="MuiTypography-root MuiTypography-caption"
          >
            <div>
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root makeStyles-iconError-4"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-2h2v2zm0-4h-2V7h2v6z"
                />
              </svg>
              1
              Â transactions have errors and will not be imported
            </div>
          </span>
        </div>
      </div>
      <div
        className="TableToolbar-spacer-16"
      />
      <div
        className="TableToolbar-actions-17"
      >
        <label
          className="MuiFormControlLabel-root ResultsToolbarComponent-chekbox-13"
        >
          <span
            className="MuiSwitch-root"
          >
            <span
              aria-disabled={false}
              className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-70 MuiSwitch-switchBase MuiSwitch-colorSecondary"
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={null}
            >
              <span
                className="MuiIconButton-label"
              >
                <span
                  className="MuiSwitch-thumb"
                />
                <input
                  checked={false}
                  className="PrivateSwitchBase-input-73 MuiSwitch-input"
                  onChange={[Function]}
                  type="checkbox"
                  value="showOnlyErrors"
                />
              </span>
              <span
                className="MuiTouchRipple-root"
              />
            </span>
            <span
              className="MuiSwitch-track"
            />
          </span>
          <span
            className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
          >
            Show only errors
          </span>
        </label>
      </div>
    </div>
    <div
      style={
        Object {
          "height": 0,
          "overflow": "visible",
          "width": 0,
        }
      }
    >
      <div
        aria-colcount={6}
        aria-rowcount={3}
        className="ReactVirtualized__Table"
        role="grid"
        style={Object {}}
      >
        <div
          className="ReactVirtualized__Table__headerRow TransactionsTableComponent-headerRow-6 TransactionsTableComponent-oddRow-9"
          role="row"
          style={
            Object {
              "height": 25,
              "overflow": "hidden",
              "paddingRight": 0,
              "width": 0,
            }
          }
        >
          <div
            aria-label="Date"
            aria-sort="descending"
            className="ReactVirtualized__Table__headerColumn ReactVirtualized__Table__sortableHeaderColumn"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="columnheader"
            style={
              Object {
                "WebkitFlex": "0 1 120px",
                "flex": "0 1 120px",
                "msFlex": "0 1 120px",
              }
            }
            tabIndex={0}
          >
            <span
              className="ReactVirtualized__Table__headerTruncatedText"
              title="Date"
            >
              Date
            </span>
            <svg
              className="ReactVirtualized__Table__sortableHeaderIcon ReactVirtualized__Table__sortableHeaderIcon--DESC"
              height={18}
              viewBox="0 0 24 24"
              width={18}
            >
              <path
                d="M7 10l5 5 5-5z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </div>
          <div
            className="ReactVirtualized__Table__headerColumn"
            role="columnheader"
            style={
              Object {
                "WebkitFlex": "1 1 200px",
                "flex": "1 1 200px",
                "msFlex": "1 1 200px",
              }
            }
          >
            <span
              className="ReactVirtualized__Table__headerTruncatedText"
              title="Description"
            >
              Description
            </span>
          </div>
          <div
            aria-label="Category"
            aria-sort="none"
            className="ReactVirtualized__Table__headerColumn ReactVirtualized__Table__sortableHeaderColumn"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="columnheader"
            style={
              Object {
                "WebkitFlex": "0 1 200px",
                "flex": "0 1 200px",
                "msFlex": "0 1 200px",
              }
            }
            tabIndex={0}
          >
            <span
              className="ReactVirtualized__Table__headerTruncatedText"
              title="Category"
            >
              Category
            </span>
          </div>
          <div
            aria-label="In"
            aria-sort="none"
            className="ReactVirtualized__Table__headerColumn ReactVirtualized__Table__sortableHeaderColumn"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="columnheader"
            style={
              Object {
                "WebkitFlex": "0 1 130px",
                "flex": "0 1 130px",
                "msFlex": "0 1 130px",
              }
            }
            tabIndex={0}
          >
            <span
              className="ReactVirtualized__Table__headerTruncatedText"
              title="In"
            >
              In
            </span>
          </div>
          <div
            aria-label="Out"
            aria-sort="none"
            className="ReactVirtualized__Table__headerColumn ReactVirtualized__Table__sortableHeaderColumn"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="columnheader"
            style={
              Object {
                "WebkitFlex": "0 1 130px",
                "flex": "0 1 130px",
                "msFlex": "0 1 130px",
              }
            }
            tabIndex={0}
          >
            <span
              className="ReactVirtualized__Table__headerTruncatedText"
              title="Out"
            >
              Out
            </span>
          </div>
          <div
            className="ReactVirtualized__Table__headerColumn"
            role="columnheader"
            style={
              Object {
                "WebkitFlex": "0 1 20px",
                "flex": "0 1 20px",
                "msFlex": "0 1 20px",
              }
            }
          >
            <span
              className="ReactVirtualized__Table__headerTruncatedText"
              title={null}
            />
          </div>
        </div>
        <div
          aria-label="grid"
          aria-readonly={true}
          className="ReactVirtualized__Grid ReactVirtualized__Table__Grid"
          onScroll={[Function]}
          role="rowgroup"
          style={
            Object {
              "WebkitOverflowScrolling": "touch",
              "boxSizing": "border-box",
              "direction": "ltr",
              "height": -25,
              "overflowX": "hidden",
              "overflowY": "auto",
              "position": "relative",
              "width": 0,
              "willChange": "transform",
            }
          }
          tabIndex={0}
        />
      </div>
    </div>
  </div>
  <hr
    className="MuiDivider-root"
  />
  <div
    className="makeStyles-formActions-2"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSecondary"
      disabled={false}
      onBlur={[Function]}
      onClick={[MockFunction]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Save Transactions
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSecondary"
      disabled={false}
      onBlur={[Function]}
      onClick={[MockFunction]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Back
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;

exports[`ImportedResults matches snapshot without errors 1`] = `
<div>
  <div
    className="TransactionsTableComponent-tableWrapper-5 makeStyles-tableWrapper-1"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular TableToolbar-root-14 MuiToolbar-gutters"
    >
      <div
        className="TableToolbar-title-18"
      >
        <div>
          <h6
            className="MuiTypography-root MuiTypography-h6"
          >
            Found 2 transactions
          </h6>
          <span
            className="MuiTypography-root MuiTypography-caption"
          >
            No errors
          </span>
        </div>
      </div>
      <div
        className="TableToolbar-spacer-16"
      />
      <div
        className="TableToolbar-actions-17"
      >
        <label
          className="MuiFormControlLabel-root ResultsToolbarComponent-chekbox-13"
        >
          <span
            className="MuiSwitch-root"
          >
            <span
              aria-disabled={false}
              className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-70 MuiSwitch-switchBase MuiSwitch-colorSecondary"
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={null}
            >
              <span
                className="MuiIconButton-label"
              >
                <span
                  className="MuiSwitch-thumb"
                />
                <input
                  checked={false}
                  className="PrivateSwitchBase-input-73 MuiSwitch-input"
                  onChange={[Function]}
                  type="checkbox"
                  value="showOnlyErrors"
                />
              </span>
              <span
                className="MuiTouchRipple-root"
              />
            </span>
            <span
              className="MuiSwitch-track"
            />
          </span>
          <span
            className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
          >
            Show only errors
          </span>
        </label>
      </div>
    </div>
    <div
      style={
        Object {
          "height": 0,
          "overflow": "visible",
          "width": 0,
        }
      }
    >
      <div
        aria-colcount={6}
        aria-rowcount={2}
        className="ReactVirtualized__Table"
        role="grid"
        style={Object {}}
      >
        <div
          className="ReactVirtualized__Table__headerRow TransactionsTableComponent-headerRow-6 TransactionsTableComponent-oddRow-9"
          role="row"
          style={
            Object {
              "height": 25,
              "overflow": "hidden",
              "paddingRight": 0,
              "width": 0,
            }
          }
        >
          <div
            aria-label="Date"
            aria-sort="descending"
            className="ReactVirtualized__Table__headerColumn ReactVirtualized__Table__sortableHeaderColumn"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="columnheader"
            style={
              Object {
                "WebkitFlex": "0 1 120px",
                "flex": "0 1 120px",
                "msFlex": "0 1 120px",
              }
            }
            tabIndex={0}
          >
            <span
              className="ReactVirtualized__Table__headerTruncatedText"
              title="Date"
            >
              Date
            </span>
            <svg
              className="ReactVirtualized__Table__sortableHeaderIcon ReactVirtualized__Table__sortableHeaderIcon--DESC"
              height={18}
              viewBox="0 0 24 24"
              width={18}
            >
              <path
                d="M7 10l5 5 5-5z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </div>
          <div
            className="ReactVirtualized__Table__headerColumn"
            role="columnheader"
            style={
              Object {
                "WebkitFlex": "1 1 200px",
                "flex": "1 1 200px",
                "msFlex": "1 1 200px",
              }
            }
          >
            <span
              className="ReactVirtualized__Table__headerTruncatedText"
              title="Description"
            >
              Description
            </span>
          </div>
          <div
            aria-label="Category"
            aria-sort="none"
            className="ReactVirtualized__Table__headerColumn ReactVirtualized__Table__sortableHeaderColumn"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="columnheader"
            style={
              Object {
                "WebkitFlex": "0 1 200px",
                "flex": "0 1 200px",
                "msFlex": "0 1 200px",
              }
            }
            tabIndex={0}
          >
            <span
              className="ReactVirtualized__Table__headerTruncatedText"
              title="Category"
            >
              Category
            </span>
          </div>
          <div
            aria-label="In"
            aria-sort="none"
            className="ReactVirtualized__Table__headerColumn ReactVirtualized__Table__sortableHeaderColumn"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="columnheader"
            style={
              Object {
                "WebkitFlex": "0 1 130px",
                "flex": "0 1 130px",
                "msFlex": "0 1 130px",
              }
            }
            tabIndex={0}
          >
            <span
              className="ReactVirtualized__Table__headerTruncatedText"
              title="In"
            >
              In
            </span>
          </div>
          <div
            aria-label="Out"
            aria-sort="none"
            className="ReactVirtualized__Table__headerColumn ReactVirtualized__Table__sortableHeaderColumn"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="columnheader"
            style={
              Object {
                "WebkitFlex": "0 1 130px",
                "flex": "0 1 130px",
                "msFlex": "0 1 130px",
              }
            }
            tabIndex={0}
          >
            <span
              className="ReactVirtualized__Table__headerTruncatedText"
              title="Out"
            >
              Out
            </span>
          </div>
          <div
            className="ReactVirtualized__Table__headerColumn"
            role="columnheader"
            style={
              Object {
                "WebkitFlex": "0 1 20px",
                "flex": "0 1 20px",
                "msFlex": "0 1 20px",
              }
            }
          >
            <span
              className="ReactVirtualized__Table__headerTruncatedText"
              title={null}
            />
          </div>
        </div>
        <div
          aria-label="grid"
          aria-readonly={true}
          className="ReactVirtualized__Grid ReactVirtualized__Table__Grid"
          onScroll={[Function]}
          role="rowgroup"
          style={
            Object {
              "WebkitOverflowScrolling": "touch",
              "boxSizing": "border-box",
              "direction": "ltr",
              "height": -25,
              "overflowX": "hidden",
              "overflowY": "auto",
              "position": "relative",
              "width": 0,
              "willChange": "transform",
            }
          }
          tabIndex={0}
        />
      </div>
    </div>
  </div>
  <hr
    className="MuiDivider-root"
  />
  <div
    className="makeStyles-formActions-2"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSecondary"
      disabled={false}
      onBlur={[Function]}
      onClick={[MockFunction]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Save Transactions
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSecondary"
      disabled={false}
      onBlur={[Function]}
      onClick={[MockFunction]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Back
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
